plugins/groq/superduper_groq/model.py

```diff
@@ -0,0 +1,87 @@
+import groq
+from groq import APIConnectionError, APIError, APIStatusError, APITimeoutError
+from superduper.components.model import APIBaseModel
+from superduper.base.query_dataset import QueryDataset
+import os
+import dataclasses as dc
+from superduper.misc.retry import Retry
+from typing import Any, Optional, Union, List, Dict
+from superduper.misc.utils import format_prompt
+
+
+retry = Retry(
+    exception_types=(APIConnectionError, APIError, APIStatusError, APITimeoutError)
+)
+
+
+class GroqAPIModel(APIBaseModel):
+    """Base Class for Groq API Models
+
+    :param groq_api_key: The API key for authenticating with the Groq API
+    """
+
+    GROQ_API_KEY: Optional[str] = os.getenv("GROQ_API_KEY")
```

Comment:
Probably best not to save the key in the attributes. That would mean it goes in the database.